core/Archive/ArchiveInvalidator.php

```diff
@@ -654,6 +654,9 @@ public function removeInvalidationsFromDistributedList($idSites, $pluginName = n
         $list = new ReArchiveList();
         $entries = $list->getAll();
 
+        // Make sure that idSites is an array to prevent typeError
+        $idSites = is_array($idSites) || $idSites === 'all' ? $idSites : (is_numeric($idSites) ? [$idSites] : []);
+
         if ($idSites === 'all') {
             $idSites = $this->getAllSitesId();
         }
```

Comment:
Switching the checks might make everything a bit simpler.

```suggestion
        if ($idSites === 'all') {
            $idSites = $this->getAllSitesId();
        }

        // Make sure that idSites is an array to prevent typeError
        $idSites = !is_array($idSites) ?  [$idSites] : $idSites;
```

Reply:
@sgiehl Thank you. Good call using the existing 'all' check. I found a test case which resulted in unexpected results with the ternary  completely simplified, so I had to have one additional check.